ArduPlane/navigation.cpp

```diff
@@ -182,22 +182,31 @@ void Plane::calc_airspeed_errors()
         }
 
 #endif // OFFBOARD_GUIDED == ENABLED
+
     } else if (flight_stage == AP_Vehicle::FixedWing::FLIGHT_LAND) {
         // Landing airspeed target
         target_airspeed_cm = landing.get_target_airspeed_cm();
-    } else if ((control_mode == &mode_auto) &&
-               (quadplane.options & QuadPlane::OPTION_MISSION_LAND_FW_APPROACH) &&
-							 ((vtol_approach_s.approach_stage == Landing_ApproachStage::APPROACH_LINE) ||
-							  (vtol_approach_s.approach_stage == Landing_ApproachStage::VTOL_LANDING))) {
-        const float land_airspeed = SpdHgt_Controller->get_land_airspeed();
-        if (is_positive(land_airspeed)) {
-            target_airspeed_cm = land_airspeed * 100;
+    } else if (control_mode == &mode_guided && new_airspeed_cm >= 0) { //DO_CHANGE_SPEED overrides onboard guided speed commands, user would have re-enter guided mode to revert
+                       target_airspeed_cm = new_airspeed_cm;
+    } else if (control_mode == &mode_auto) {
+        if ((quadplane.options & QuadPlane::OPTION_MISSION_LAND_FW_APPROACH) &&
+			   ((vtol_approach_s.approach_stage == Landing_ApproachStage::APPROACH_LINE) ||
+			     (vtol_approach_s.approach_stage == Landing_ApproachStage::VTOL_LANDING))) {
+                     const float land_airspeed = SpdHgt_Controller->get_land_airspeed();
+                     if (is_positive(land_airspeed)) {
+                         target_airspeed_cm = land_airspeed * 100;
+                     } else {
+                         // fallover to normal airspeed
+                         target_airspeed_cm = aparm.airspeed_cruise_cm;
+                     }
         } else {
-            // fallover to normal airspeed
-            target_airspeed_cm = aparm.airspeed_cruise_cm;
+            // normal AUTO mode and new_airspeed variable was set by DO_CHANGE_SPEED command while in AUTO mode
+            if (new_airspeed_cm !=-1) {
```

Comment:
check for > 0

Reply:
can we change all the checks on new_airspeed_cm to be > 0 please

Reply:
fixed